app/index.js

```diff
@@ -81,6 +81,7 @@ module.exports = class GeneratorLicense extends Generator {
         name: 'license',
         message: 'Which license do you want to use?',
         default: this.options.defaultLicense,
+        when: this.options.license == null || licenses.find(x => x.value === this.options.license) === undefined,
```

Comment:
I believe this would be sufficient and do same thing:

    when: !this.options.license